oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/config/OneAndOneComputeServiceContextModule.java

```diff
@@ -0,0 +1,179 @@
+      }).to(OneandoneComputeServiceAdapter.class);
+
+      bind(TemplateBuilderImpl.class).to(ArbitraryCpuRamTemplateBuilderImpl.class);
+
+      bind(new TypeLiteral<Function<Server, NodeMetadata>>() {
+      }).to(ServerToNodeMetadata.class);
+
+      bind(new TypeLiteral<Function<org.apache.jclouds.oneandone.rest.domain.Image, Image>>() {
+      }).to(ImageToImage.class);
+
+      bind(new TypeLiteral<Function<Hdd, Volume>>() {
+      }).to(HddToVolume.class);
+
+      bind(new TypeLiteral<Function<HardwareFlavour, Hardware>>() {
+      }).to(HardwareFlavourToHardware.class);
+   }
+
+   @Provides
+   @Singleton
+   @Named(POLL_PREDICATE_SERVER_RUNNING)
+   Predicate<Server> provideServerAvailablePredicate(final OneAndOneApi api, ComputeConstants constants) {
+      return retry(new ServerAvailablePredicate(api),
+              constants.pollTimeout(), constants.pollPeriod(), constants.pollMaxPeriod(), TimeUnit.SECONDS);
+   }
+
+   @Provides
+   @Singleton
+   @Named(POLL_PREDICATE_SERVER)
+   Predicate<Server> ServerRunningPredicate(final OneAndOneApi api, ComputeConstants constants) {
+      return retry(new ServerRunningPredicate(api, Types.ServerState.POWERED_ON),
+              constants.pollTimeout(), constants.pollPeriod(), constants.pollMaxPeriod(), TimeUnit.SECONDS);
+   }
+
+   static class ServerAvailablePredicate implements Predicate<Server> {
+
+      private final OneAndOneApi api;
+
+      public ServerAvailablePredicate(OneAndOneApi api) {
+         this.api = checkNotNull(api, "api must not be null");
+      }
+
+      @Override
+      public boolean apply(Server server) {
+         checkNotNull(server, "Server");
+         server = api.serverApi().get(server.id());
+         if ((server.status().state() != Types.ServerState.POWERED_OFF
+                 && server.status().state() != Types.ServerState.POWERED_ON)
+                 || server.status().percent() != 0) {
+            return false;
+         } else {
+            return true;
+         }
+      }
+
+   }
+
+   static class ServerRunningPredicate implements Predicate<Server> {
+
+      private final OneAndOneApi api;
+      private final Types.ServerState expectedState;
+
+      public ServerRunningPredicate(OneAndOneApi api, Types.ServerState expectedState) {
+         this.api = checkNotNull(api, "api must not be null");
+         this.expectedState = checkNotNull(expectedState, "expectedState must not be null");
+      }
+
+      @Override
+      public boolean apply(Server serverRef) {
+
+         checkNotNull(serverRef, "serverRef");
+         //give time for the operation to actually start
+         Uninterruptibles.sleepUninterruptibly(10, TimeUnit.SECONDS);
+         Server server = api.serverApi().get(serverRef.id());
+
+         if (server == null) {
+            return false;
+         }
+         return server.status().toString().equals(expectedState.toString());
+      }
+
+   }
+
+   @Singleton
+   public static class ComputeConstants {
```

Comment:
Get rid of this class and use the existing [PollPeriod](https://github.com/jclouds/jclouds/blob/master/compute/src/main/java/org/jclouds/compute/reference/ComputeServiceConstants.java#L73-L82) and [Timeouts](https://github.com/jclouds/jclouds/blob/master/compute/src/main/java/org/jclouds/compute/reference/ComputeServiceConstants.java#L94-L105).